tao_compiler/mlir/disc/transforms/disc_collective_ops_rewriter.cc

```diff
@@ -55,6 +55,24 @@ std::optional<std::string> ReductionKindToString(ReductionKind kind) {
   return std::nullopt;
 }
 
+bool EnableAsyncCollective(Operation* op) {
+  if (llvm::isa<mhlo::AllReduceOp>(op)) {
+    if (const char* env_p = std::getenv("ENABLE_ASYNC_ALL_REDUCE")) {
```

Comment:
make all collective ops to be async? Do we need to separate sync and async collective ops ?

Reply:
Just mimicking [openXLA's setting and behaviour](https://github.com/openxla/xla/blob/4c566c945ad38e69bc71b5ee67b741551187e011/xla/xla.proto#L388)

Reply:
Fixed.